src/arfsdao.ts

```diff
@@ -636,3 +723,70 @@ export class ArFSPublicDriveBuilder {
 		throw new Error('Invalid drive state');
 	}
 }
+
+export class ArFSPublicFolder extends ArFSEntity implements ArFSFileFolderEntity {
+	constructor(
+		readonly appName: string,
+		readonly appVersion: string,
+		readonly arFS: string,
+		readonly contentType: string,
+		readonly driveId: string,
+		readonly entityType: string,
+		readonly name: string,
+		readonly txId: string,
+		readonly unixTime: number,
+		readonly parentFolderId: string,
+		readonly entityId: string,
+		readonly lastModifiedDate: number
+	) {
+		super(appName, appVersion, arFS, contentType, driveId, entityType, name, 0, txId, unixTime);
+	}
+}
+
+export class ArFSPublicFolderBuilder {
+	appName?: string;
+	appVersion?: string;
+	arFS?: string;
+	contentType?: ContentType;
+	driveId?: DriveID;
+	entityType?: EntityType;
+	name?: string;
+	txId?: TransactionID;
+	unixTime?: number;
+	parentFolderId?: string;
+	entityId?: string;
+	lastModifiedDate?: number;
+
+	build(): ArFSPublicFolder {
+		if (
+			this.appName?.length &&
+			this.appVersion?.length &&
+			this.arFS?.length &&
+			this.contentType?.length &&
+			this.driveId?.length &&
+			this.entityType?.length &&
+			this.name?.length &&
+			this.txId?.length &&
+			this.unixTime &&
+			this.entityId?.length
+			// FIXME: Is the Last-Modified-Date missing sometimes?
+			// this.lastModifiedDate
+		) {
+			return new ArFSPublicFolder(
+				this.appName,
+				this.appVersion,
+				this.arFS,
+				this.contentType,
+				this.driveId,
+				this.entityType,
+				this.name,
+				this.txId,
+				this.unixTime,
+				this.parentFolderId || '',
```

Comment:
This one can only happen with the root folder of a drive, correct?

If so, maybe we should put something useful to users here instead of an empty string. Like `"root folder"`?

Reply:
Makes sense. Applied at b961a00493fbc13e7ecfd66abf43123ab90dd5f3